tests/source/test_csv.py

```diff
@@ -1,19 +1,68 @@
 # SPDX-License-Identifier: MIT
 # Copyright (c) 2019 Intel Corporation
 import unittest
+import tempfile
 
 from dffml.source.file import FileSourceConfig
-from dffml.source.csv import CSVSource
+from dffml.source.csv import CSVSource, CSVSourceConfig
 from dffml.util.testing.source import FileSourceTest
 from dffml.util.asynctestcase import AsyncTestCase
+from dffml.repo import Repo
+from dffml.util.cli.arg import parse_unknown
 
 
 class TestCSVSource(FileSourceTest, AsyncTestCase):
     async def setUpSource(self):
-        return CSVSource(FileSourceConfig(filename=self.testfile))
+        return CSVSource(CSVSourceConfig(filename=self.testfile))
 
     @unittest.skip("Labels not implemented yet for CSV files")
     async def test_label(self):
         """
         Labels not implemented yet for CSV files
         """
+
+    def test_config_readonly_default(self):
+        config = CSVSource.config(
+            parse_unknown("--source-file-filename", "feedface")
+        )
+        self.assertEqual(config.filename, "feedface")
+        self.assertEqual(config.label, "unlabeled")
+        self.assertEqual(config.key, None)
+        self.assertFalse(config.readonly)
+
+    def test_config_readonly_set(self):
+        config = CSVSource.config(
+            parse_unknown(
+                "--source-file-filename",
+                "feedface",
+                "--source-file-label",
+                "default-label",
+                "--source-file-key",
+                "SourceURLColumn",
+                "--source-file-readonly",
+            )
+        )
+        self.assertEqual(config.filename, "feedface")
+        self.assertEqual(config.label, "default-label")
+        self.assertEqual(config.key, "SourceURLColumn")
+        self.assertTrue(config.readonly)
+
+    def config(self, filename, label="unlabeled", key=None, readonly=False):
```

Comment:
Looks like this `config()`  method is never called, can you please remove it? https://codecov.io/gh/intel/dffml/pull/102/src/tests/source/test_csv.py#L51

Reply:
Done :+1: